packages/eventual-send/src/index.d.ts

```diff
@@ -18,6 +18,20 @@ export type EOnly<T> = T extends (...args: infer P) => infer R
     >
   : ERef<T>;
 
+type FilteredKeys<T, U> = {
+  [P in keyof T]: T[P] extends U ? P : never;
+}[keyof T];
+
+type DataOnly<T> = Omit<T, FilteredKeys<T, Function>>;
+type FunctionOnly<T> = Pick<T, FilteredKeys<T, Function>> &
+  (T extends Function ? (...args: Parameters<T>) => ReturnType<T> : {});
+
+interface Remotable<T> {
+  __Remote__: T;
```

Comment:
How would that work? I imagine it would make the apparent type harder to interpret.

Reply:
Something like 

```ts
export declare const remoteSymbol: unique symbol;

interface Remotable<T> {
  [remoteSymbol]: T;
}
```

It'd avoid making `__Remote__` a string property of the `Remotable`.

Reply:
Neat! I thought it would put some more confusing string into the type, but it seems to be strictly better :) 
```
export declare const remoteSymbol: unique symbol;

interface Remotable<T> {
  [remoteSymbol]: T;
  __Remote__: T;
}

const obj = {a: 1, b: 2};

const robj = obj as unknown as Remotable<typeof obj>;
const r1 = robj.__Remote__; // passes typecheck, which is undesired
const r2 = robj[remoteSymbol]; // also passes typecheck but at least it's not a member of robj
```

Reply:
Correct, and we may get away with not exporting the `remoteSymbol` declaration, so actual code wouldn't even be able to write `robj[remoteSymbol]`